File: data/readable_format/fortran/file2269.f90
program test_sqdist
    implicit none
    real*8 :: result
    real*8 :: x1, y1, z1, x2, y2, z2
    integer :: ind, MAXROT
    real*8 :: rotmat(3, 3, 3)

    ! Initialize test values
    x1 = 1.0
    y1 = 2.0
    z1 = 3.0
    x2 = 4.0
    y2 = 5.0
    z2 = 6.0
    ind = 1
    MAXROT = 3

    ! Initialize the rotation matrix
    rotmat(1,1,1) = 1.0
    rotmat(1,1,2) = 0.0
    rotmat(1,1,3) = 0.0
    rotmat(1,2,1) = 0.0
    rotmat(1,2,2) = 1.0
    rotmat(1,2,3) = 0.0
    rotmat(1,3,1) = 0.0
    rotmat(1,3,2) = 0.0
    rotmat(1,3,3) = 1.0

    ! Call the function
    result = sqdist(x1, y1, z1, x2, y2, z2, ind, MAXROT, rotmat)

    ! Print the result
    print *, 'Squared distance:', result

contains

    real*8 function sqdist(x1, y1, z1, x2, y2, z2, ind, MAXROT, rotmat)
        real*8 :: x1, y1, z1, x2, y2, z2
        integer :: ind, MAXROT
        real*8 :: rotmat(MAXROT, 3, 3)
        real*8 :: cont, dx, dy, dz
        integer :: i

        dx = dble(x1 - x2)
        dy = dble(y1 - y2)
        dz = dble(z1 - z2)
        sqdist = 0.0
        do i = 1, 3
            cont = rotmat(ind, i, 1) * dx &
                 + rotmat(ind, i, 2) * dy &
                 + rotmat(ind, i, 3) * dz
            sqdist = sqdist + cont * cont
        end do
    end function sqdist

end program test_sqdist

File: data/readable_format/cpp/file2269.cpp
#include <gtest/gtest.h>
#include <vector>
#include <cmath> // Include for std::pow

double sqdist(double x1, double y1, double z1, double x2, double y2, double z2, int ind, int MAXROT, const std::vector<std::vector<std::vector<double>>>& rotmat) {
    double dx = x1 - x2;
    double dy = y1 - y2;
    double dz = z1 - z2;
    double sqdist = 0.0;

    for (int i = 0; i < 3; ++i) {
        double cont = rotmat[ind][i][0] * dx
                    + rotmat[ind][i][1] * dy
                    + rotmat[ind][i][2] * dz;
        sqdist += cont * cont;
    }

    return sqdist;
}

TEST(SqDistTest, BasicTest) {
    double x1 = 1.0, y1 = 2.0, z1 = 3.0;
    double x2 = 4.0, y2 = 5.0, z2 = 6.0;
    int ind = 0;
    int MAXROT = 3;

    std::vector<std::vector<std::vector<double>>> rotmat(MAXROT, std::vector<std::vector<double>>(3, std::vector<double>(3, 0.0)));

    // Initialize the rotation matrix
    rotmat[0][0][0] = 1.0; rotmat[0][0][1] = 0.0; rotmat[0][0][2] = 0.0;
    rotmat[0][1][0] = 0.0; rotmat[0][1][1] = 1.0; rotmat[0][1][2] = 0.0;
    rotmat[0][2][0] = 0.0; rotmat[0][2][1] = 0.0; rotmat[0][2][2] = 1.0;

    double result = sqdist(x1, y1, z1, x2, y2, z2, ind, MAXROT, rotmat);

    // Expected result calculation
    double expected = std::pow(x1 - x2, 2) + std::pow(y1 - y2, 2) + std::pow(z1 - z2, 2);

    EXPECT_DOUBLE_EQ(result, expected);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}